modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/extras/SourceConfirmedTextQuery.java

```diff
@@ -438,7 +439,13 @@ private MemoryIndex getOrCreateMemoryIndex() throws IOException {
                     if (value == null) {
                         continue;
                     }
-                    cacheEntry.memoryIndex.addField(field, value.toString(), indexAnalyzer);
+                    String valueStr;
```

Comment:
I don't follow completely why this change is need. I think this uses value fetcher, which then uses source? Synthetic source should still serialize as string based fields as strings?

Reply:
Since we're creating the stored field with a `BytesRef` instead of a `String` now, it deserializes into a `BytesRef`. Then, when `toString` was called here, it would return the hexadecimal representation of the underlying bytes instead of the actual string value.

This update was needed to make the `MatchOnlyTextFieldMapperTests#testPhraseQuerySyntheticSource()` test pass. Without this update, the query would return 0 hits, failing the test.